thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/dao/GenericPojoDao.java

```diff
@@ -68,6 +72,22 @@ public GenericPojoDao(final DatabaseService databaseService,
         "Entity Metadata is inconsistent!");
   }
 
+  private static void validateLimitOffset(final DaoFilter filter) {
```

Comment:
I think you can just call this `validate`

Please use `Preconditions.checkArgument()`, not `ThirdEyeException`. The latter is primarily a framework for communicating with UI.